File: samples/SampleVBProject/SampleModule.vb
Module SampleModule

    Public Function SafeDivide(a As Integer, b As Integer) As Integer
        If b Is Nothing Then Return 0
        Return a \ b
    End Function

    Public Function CalcPower(base_ As Double, exp As Double) As Double
        Return base_ ^ exp
    End Function

    Public Function CompareIgnoreCase(a As String, b As String) As Boolean
        Return String.Compare(a, b, True) = 0
    End Function

    Public Function GetSetting() As String
        Return My.Settings.AppTitle
    End Function

    Public Sub ProcessArray(ByRef arr() As Integer, newSize As Integer)
        ReDim Preserve arr(newSize)
    End Sub

    Public Function UseIIf(x As Integer) As String
        Return IIf(x > 0, "positive", "non-positive")
    End Function

    Public Sub HandleError()
        On Error GoTo ErrHandler
        Dim x As Integer = 1 / 0
        Exit Sub
ErrHandler:
        Console.WriteLine(Err.Description)
    End Sub

End Module

File: samples/SampleVBProject/SampleModule.cs

    public static int SafeDivide(int a, int b)
    {
        if (b is null)
            return 0;
        return a / b;
    }



    public static double CalcPower(double base_, double exp)
    {
        return Math.Pow(base_, exp);
    }



    public static bool CompareIgnoreCase(string a, string b)
    {
        return string.Compare(a, b, true) == 0;
    }



    public static string GetSetting()
    {
        return Properties.Settings.Default.AppTitle;
    }



    public static void ProcessArray(ref int[] arr, int newSize)
    {
        Array.Resize(ref arr, newSize + 1);
    }



    public static string UseIIf(int x)
    {
        return (/* ⚠ VBMigrator: IIf evalúa ambos brazos en VB; ternario ?: no lo hace */ x > 0 ? "positive" : "non-positive");
    }



    public static void HandleError()
    {
        try
        {
            int x = (int)Math.Round(1d / 0d);
            return;
        }
        catch
        {

            Console.WriteLine(Err.Description);
        }
    }
